pkg/registry/core/pod/storage/eviction.go

```diff
@@ -292,6 +293,39 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 	return &metav1.Status{Status: metav1.StatusSuccess}, nil
 }
 
+func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string, validation rest.ValidateObjectFunc, options *metav1.DeleteOptions) error {
+	if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+		if pod, err := getPod(r, ctx, name); err != nil {
+			return err
+		} else if pod != nil {
+			conditionAppender := func(_ context.Context, newObj, _ runtime.Object) (runtime.Object, error) {
```

Comment:
You can reduce nesting like this:

```suggestion
		pod, err := getPod(r, ctx, name);
		if err != nil {
			return err
		} 
		conditionAppender := func(_ context.Context, newObj, _ runtime.Object) (runtime.Object, error) {
```